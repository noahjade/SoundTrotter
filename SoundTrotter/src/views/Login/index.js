import React from "react";
import style from "./style.module.scss";
import DefaultButton from "../../components/defaultButton";
import { loginUrl } from "../../utils/spotifyService";

function Login() {
  return (
    <div className={style.login}>
      <a href={loginUrl}>
        <DefaultButton text="Login with spotify" />
      </a>
    </div>
  );
}

export default Login;
