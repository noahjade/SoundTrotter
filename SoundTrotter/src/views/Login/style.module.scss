.login {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  img {
    width: 50vw;
    margin-bottom: 20px;
  }
}
