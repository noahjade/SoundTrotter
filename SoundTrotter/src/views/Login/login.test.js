import renderer from "react-test-renderer";
import Login from ".";

test("Correctly renders login view", () => {
  const view = renderer.create(<Login />);
  expect(view).toMatchSnapshot();
});
