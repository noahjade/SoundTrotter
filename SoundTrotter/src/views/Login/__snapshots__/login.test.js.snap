// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders login view 1`] = `
<div
  className="login"
>
  <a
    href="https://accounts.spotify.com/authorize?client_id=376c736575c0456f8215eebec0eee1d2&redirect_uri=http://localhost:3000/&scope=user-read-currently-playing%20user-read-recently-played%20user-read-playback-state%20user-top-read%20user-modify-playback-state%20streaming%20user-read-email%20user-read-private%20user-library-read%20playlist-read-collaborative%20playlist-read-private%20playlist-read-collaborative%20playlist-modify-public%20playlist-modify-private&response_type=code&show_dialog=true"
  >
    <button
      className="btn"
    >
      Login with spotify
    </button>
  </a>
</div>
`;
